src/auth/views/Login.tsx

```diff
@@ -50,6 +55,8 @@ const LoginView: React.FC<LoginViewProps> = ({ params }) => {
   };
 
   const handleRequestExternalAuthentication = async (pluginId: string) => {
+    setFallbackUri(location.pathname);
```

Comment:
Can we not add it as a query parameter to `redirectUri` below?
```
      redirectUri: urlJoin(
        window.location.origin,
        APP_MOUNT_URI === APP_DEFAULT_URI ? "" : APP_MOUNT_URI,
        loginCallbackPath,
        `?fallback=${location.pathname}`
      )
```

Reply:
I tried this solution, it would require backend changes as backend's response includes only the callback url, additionally auth0 is strict on the callback path - from what i've seen it cannot be dynamic. Auth0, in our case, recommends storing the path we call `fallback` in the browser storage.

Reply:
In that case let's just put it in the url. `Login` view uses a non-exact path match, so it will match all pathnames that look like this: `/login/*`. We can leverage this and append our url directly to the main pathname:
```
   redirectUri: urlJoin(
       location.origin,
        APP_MOUNT_URI === APP_DEFAULT_URI ? "" : APP_MOUNT_URI,
        loginCallbackPath,
       location.pathname
      )
```

Then we can extract it from the url when `handleExternalAuthentication` is called. No changes are needed to path detection as:
```
const isCallbackPath = location.pathname.includes(loginCallbackPath);
```
will continue to work as it used to

Reply:
The solutions work the same as in your previous comment - passing `location.pathname` to the `requestLoginByExternalPlugin`. `requestLoginByExternalPlugin` sends it off to the auth0 and it won't accept an uri that is not predefined in the auth0 dashboard - it cannot be dynamic. 
It causes the following error:
```
Callback URL mismatch.
The provided redirect_uri is not in the list of allowed callback URLs.
```
It would work on our side but not on the auth0 side, therefore we will have to stick with storing the pathname in browser storage as it is recommended by auth0 (eg. here https://community.auth0.com/t/how-do-i-set-up-a-dynamic-allowed-callback-url/60268 but most importantly here https://auth0.com/docs/secure/attack-protection/state-parameters)